### PR TITLE
Allow an optional background image

### PR DESCRIPTION
**Pull request by gbalbuena**
captura de pantalla 2016-09-19 a las 23 37 23

Implements #5

### CODE REVIEW

**Comment by gbalbuena** [comment]
cc: @strzibny @drhenner

Suggestion or improvements are welcome!

File: lib/invoice_printer/document.rb [hunk 1]
```diff
@@ -81,7 +81,8 @@ class InvalidInput < StandardError; end
                 :account_swift,
                 # Collection of InvoicePrinter::Invoice::Items
                 :items,
-                :note
+                :note,
+                :background
```
Comment (strzibny) [inline lib/invoice_printer/document.rb]: We actually don't need it in here. Background will be set only for `PDFDocument` (same as with logo for instance).

File: test/background_test.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  def test_background_print
+    assert_silent do
+      InvoicePrinter.print(document: @invoice, file_name: @file_name, background: './examples/background.jpg')
+      InvoicePrinter.print(document: @invoice, file_name: @file_name, background: nil)
+      InvoicePrinter.print(document: @invoice, file_name: @file_name)
+    end
+  end
+
+  def test_missing_background_raises_an_exception
+    invoice = InvoicePrinter::Document.new(default_document_params)
```
Comment (strzibny) [inline test/background_test.rb]: This is not used in the test.

File: test/background_test.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+require 'test_helper'
+
+class BackgroundTest < Minitest::Test
+  include InvoicePrinterHelpers
+
+  def setup
+    @file_name = 'deleteme.pdf'
```
Comment (strzibny) [inline test/background_test.rb]: Let's avoid tests needed this for now. I have in mind integration tests that will actually compare the output/examples and that the features that way (so it's not just about exceptions). So test only InvoicePrinter.render and feel free to not test InvoicePrinter.print for this.

**Comment by strzibny** [comment]
Thanks, I noted a few things inline. Also please update README :).

File: test/background_test.rb (hunk 2 above)
Comment (gbalbuena) [inline test/background_test.rb]: Removed

File: test/background_test.rb (hunk 3 above)
Comment (gbalbuena) [inline test/background_test.rb]: Done

File: lib/invoice_printer/document.rb (hunk 1 above)
Comment (gbalbuena) [inline lib/invoice_printer/document.rb]: Good catch, this was my initial approach then I figure out what you describe

**Comment by gbalbuena** [comment]
You're welcome
I think it's good to go now

**Comment by gbalbuena** [comment]
@strzibny Do u think we should increase the version because of this new feature?

**Comment by strzibny** [comment]
@gbalbuena I will merge during the weekend and release a new version, no worries.

File: test/background_test.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+class BackgroundTest < Minitest::Test
+  include InvoicePrinterHelpers
+
+  def setup
+    @invoice = InvoicePrinter::Document.new(default_document_params)
+  end
+
+  def test_background_render
+    assert_silent do
```
Comment (strzibny) [inline test/background_test.rb]: `assert_silent` is too strict and will fail when Prawn libraries output warnings. This test would be enough without this call e.g. remove `do...end` block just keep the calls to `#render` themselves.

**Pull request merged**